core/src/test/java/io/jenkins/pluginhealth/scoring/probes/SecurityScanGithubWorkflowProbeTest.java

```diff
@@ -0,0 +1,196 @@
+        return spy(SecurityScanGithubWorkflowProbe.class);
+    }
+
+    @Test
+    void shouldBeAbleToDetectRepositoryWithNoGitHubWorkflowConfigured() throws IOException {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final SecurityScanGithubWorkflowProbe probe = getSpy();
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        when(ctx.getScmRepository()).thenReturn(Files.createTempDirectory("foo"));
+
+        final ProbeResult result = probe.apply(plugin, ctx);
+        assertThat(result.status()).isEqualTo(ResultStatus.FAILURE);
+        assertThat(result.message()).isEqualTo("GitHub workflow directory could not be found in the plugin");
+    }
+
+    @Test
+    void shouldBeAbleToDetectRepositoryWithNoSecurityScanConfigured() throws IOException {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final SecurityScanGithubWorkflowProbe probe = getSpy();
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        final Path repo = Files.createTempDirectory("foo");
+        Files.createDirectories(repo.resolve(".github/workflows"));
+        when(ctx.getScmRepository()).thenReturn(repo);
+
+        final ProbeResult result = probe.apply(plugin, ctx);
+        assertThat(result.status()).isEqualTo(ResultStatus.FAILURE);
+        assertThat(result.message()).isEqualTo("GitHub workflow security scan is not configured in the plugin");
+    }
+
+    @Test
+    void shouldFailIfJobsIsNotConfiguredInGitHubWorkflow() throws IOException {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final SecurityScanGithubWorkflowProbe probe = getSpy();
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        final Path repo = Files.createTempDirectory("foo");
+        Path workflowPath = Files.createDirectories(repo.resolve(".github/workflows"));
+        final Path workflowFile = Files.createFile(workflowPath.resolve("jenkins-security-scan.yaml"));
+
+        Files.write(workflowFile, List.of(
+            "name: Test Security Scan Job",
+            "nojobs-configured:",
+            "  security-scan-name:",
+            "    uses: jenkins-infra/jenkins-security-scan/.github/workflows/jenkins-security-scan.yaml"
+        ));
```

Comment:
Same as before. This is creating an invalid file syntax. This is not the way you should do it.
To be honest, when I read the test, I thought this test was vastly incorrect as you are creating a YAML file in the correct folder with the workflow we are looking for, so we should expect a `ResultStatus.SUCCESS`. 

```suggestion
        Files.write(workflowFile, List.of(
            "name: Test Security Scan Job",
            "jobs:",
            "  security-scan-name:",
            "    uses: this-is-not-the-workflow-we-are-looking-for"
        ));
```